tests/python/relax/test_transform_to_mixed_precision.py

```diff
@@ -0,0 +1,391 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import tvm
+from tvm import relax
+from tvm.relax.transform import ToMixedPrecision
+from tvm.script.parser import ir as I, relax as R
+
+
+def _assert_test(input, expected):
+    mod = ToMixedPrecision()(input)
+    tvm.ir.assert_structural_equal(mod, expected)
+
+
+def test_conv2d():
+    @I.ir_module
+    class input:
+        @R.function
+        def main(
+            x: R.Tensor((2, 3, 28, 28), "float32"), w: R.Tensor((4, 3, 3, 3), "float32")
+        ) -> R.Tensor(None, "float32", ndim=4):
+            gv: R.Tensor((2, 4, 26, 26), "float32") = R.nn.conv2d(x, w, out_dtype="float32")
+            return gv
+
+    @I.ir_module
+    class expected:
+        @R.function
+        def main(
+            x: R.Tensor((2, 3, 28, 28), dtype="float32"), w: R.Tensor((4, 3, 3, 3), dtype="float32")
+        ) -> R.Tensor((2, 4, 26, 26), dtype="float32"):
+            # block 0
+            gv: R.Tensor((2, 3, 28, 28), dtype="float16") = R.astype(x, dtype="float16")
```

Comment:
`astype` is a bit tricky IMO. How about `R.cast`? cc @tqchen

Reply:
It changed from `cast` to `astype` here https://github.com/mlc-ai/relax/pull/82.

Reply:
I prefer `y = R.cast(x, dtype="float16")` or `y = x.astype(dtype="float16")`. `R.astype` is a bit unacceptable for me :(

Reply:
cc @tqchen

Reply:
Previously the change is to align the API specification in https://data-apis.org/array-api/latest/API_specification/generated/array_api.astype.html. Alignment does bring clarity, while in this case `R.cast` indeed seems clearer than `R.astype`, which intuitively means “casting `R` into blabla.”

Reply:
I personally preferred cast before. But after seein the data api specfication(which is the defined common standard) I now agree that astype is a good choice here mainly to not form an opinion